core/src/main/scala/org/bitcoins/core/script/arithmetic/ArithmeticInterpreter.scala

```diff
@@ -111,7 +116,11 @@ sealed abstract class ArithmeticInterpreter {
     require(program.script.headOption.contains(OP_BOOLOR),
             "Script top must be OP_BOOLOR")
     performBinaryBooleanOperation(program, (x, y) => {
-      if (x == y && (x == ScriptNumber.zero || x == OP_0)) false else true
+      if (x == y && (x == ScriptNumber.zero
+          || x == OP_0
+          || x == ScriptNumber.negativeZero
+          || x.toLong == 0)) false
```

Comment:
nit: I think this is long enough now to justify a util method. This will simplify the operation on line 104-108 to
```
SomethingUtil.isZeroOrFalse(x) || SomethingUtil.isZeroOrFalse(y)
```